ledger/participant-integration-api/src/main/scala/platform/configuration/InitialLedgerConfiguration.scala

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.configuration
+
+import java.time.Duration
+
+import com.daml.ledger.configuration.Configuration
+
+/** Instructions on how to initialize an empty ledger, without a configuration.
+  *
+  * A configuation is only submitted if one is not detected on the ledger.
+  *
+  * @param configuration         The initial ledger configuration to submit.
+  * @param delayBeforeSubmitting The delay until the participant tries to submit a configuration.
+  */
+final case class InitialLedgerConfiguration(
```

Comment:
I think I am confused by the multiple meanings of "configuration", as it is both "the result of configuring" and "the action of configuring", isn't it? Not sure what I can suggest though, to clarify that this is information about how to do it; `LedgerSetupInfo` or something like that?

Reply:
I'm not sure `Info` helps.

I have tried to restrict the word "configuration" to only data objects representing the ledger configuration, whether the current configuration or a potential future one. The exception here is `CurrentLedgerConfiguration`, which I've renamed in some follow-up changes.